Automattic/vip-workflow-plugin
# Update the design for the required fields on a custom status

## Description

This PR is meant to update the design for configuring the required fields on a status, using the mockups proposed by @jarekmorawski. Check out #44 for an example of this.

The changes I've made:

- Implement a radio control for turning the user restriction on and off.
- Change the labels to be clearer and drop the help labels when they aren't necessary.

Note: The rules engine is not in place for editorial metadata just yet. Once it is, that section would change a bit more so keep that in mind when reviewing this.

![Screenshot 2024-10-02 at 12 07 35 pm](https://github.com/user-attachments/assets/499cd5ca-cc12-442a-ae6e-aab36832c29c)
![Screenshot 2024-10-02 at 12 07 45 pm](https://github.com/user-attachments/assets/bf00f015-0dd8-47f3-bb0a-de8b55412eb8)
![Screenshot 2024-10-02 at 12 07 52 pm](https://github.com/user-attachments/assets/b8ff8508-5891-4d06-8dfa-a00427a014fa)
![Screenshot 2024-10-02 at 12 07 58 pm](https://github.com/user-attachments/assets/54c9d783-1f48-437d-bca1-d7a855ef4018)
![Screenshot 2024-10-02 at 12 08 37 pm](https://github.com/user-attachments/assets/d62f69ca-6ed9-4462-8af7-7eea27974e1f)

## Steps to Test

- Update and Create a status with editorial metadata fields, as well as users to see the difference.
